### PR TITLE
Phr 588

### PR DESCRIPTION
**Pull request by Nexengineer**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 4
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/user/model/PatientName.java  5
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java  3
- src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5819553&bid=19049597)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/user/model/PatientName.java  5
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java  3
- src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5819553&bid=19049597)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/user/model/PatientName.java  5
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java  3
- src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5819553&bid=19049597)

File: src/main/java/in/projecteka/consentmanager/user/model/PatientName.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@Getter
+public class PatientName {
+    private final String first;
+    private final String middle;
+    private final String last;
+
+    public String createFullName(){
+        var fullName = first;
+
+        if (middle != null && !middle.isEmpty()){
```
Comment (dheeraj-p) [inline src/main/java/in/projecteka/consentmanager/user/model/PatientName.java:18]: Use `StringUtils.isEmpty`.

File: src/main/java/in/projecteka/consentmanager/user/model/PatientName.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            fullName += " " + middle;
+        }
+
+        if (last != null && !last.isEmpty()){
```
Comment (dheeraj-p) [inline src/main/java/in/projecteka/consentmanager/user/model/PatientName.java:22]: Here too.

File: src/main/java/in/projecteka/consentmanager/user/model/PatientName.java (hunk 1 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/consentmanager/user/model/PatientName.java]: Done

File: src/main/java/in/projecteka/consentmanager/user/model/PatientName.java (hunk 2 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/consentmanager/user/model/PatientName.java]: Done

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/user/model/PatientName.java  3
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java  3
- src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5819553&bid=19049597)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/user/model/PatientName.java  3
         

Clones added
============
- src/test/java/in/projecteka/consentmanager/user/UserServiceTest.java  2
- src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5819553&bid=19049597)

**Pull request merged**